apps/meteor/ee/app/livechat-enterprise/server/business-hour/Multiple.ts

```diff
@@ -200,7 +200,16 @@ export class MultipleBusinessHoursBehavior extends AbstractBusinessHourBehavior
 				return false;
 			}
 
-			const activeBusinessHoursForAgent = departmentsWithActiveBH.map(({ businessHourId }) => businessHourId);
+			const activeBusinessHoursForAgent = departmentsWithActiveBH.reduce(
```

Comment:
Instead of reduce, you can chain a .filter to remove "empty" items from the array 👀

Reply:
*BUT*, this way we just had to iterate once? 🙈 I am very critical that way lol.

Reply:
Yeah, this is a bit like "pick your poison". For this case, departments with active BH won't be an insane list that would hang the server 👀 most operations would have this list in the orders of 10-99 maybe. So, not sure how much load would an array of strings produce 🤔 

We can keep the reduce, just, don't want to sacrifice readability for optimization 👀 .